packages/sitecore-jss/src/data-fetcher.ts

```diff
@@ -36,33 +37,17 @@ export class ResponseError extends Error {
   }
 }
 
-/**
- * @param {HttpResponse<T>} response the response to check
- * @throws {ResponseError} if response code is not ok
- */
-export function checkStatus<T>(response: HttpResponse<T>) {
```

Comment:
I think we still need this, any reason why you remove it?

Reply:
I don't think we need it anymore. I've refactored the native-fetcher to improve error handling. The `statusCheck` function was likely created because fetch doesn't reject promises for status codes above `300`. However, since we already check `response.ok` in the native-fetcher, it serves the same purpose."